browser/playlist/playlist_tab_helper.cc

```diff
@@ -122,23 +125,26 @@ void PlaylistTabHelper::UpdateSavedItemFromCurrentContents() {
   // Should we keep a map(url, [item_id, ... , item_id]]) in PlaylistService for
   // perf improvement? We'll see this really matters.
 
-  // This is a synchronous call, so it's safe to bind this without weak
-  // reference.
-  auto on_get_all_items = base::BindOnce(
-      [](PlaylistTabHelper* helper, std::vector<mojom::PlaylistItemPtr> items) {
-        const auto& current_url = helper->web_contents()->GetVisibleURL();
-        for (const auto& item : items) {
-          if (item->page_source != current_url) {
-            continue;
-          }
-
-          helper->saved_items_.push_back(item->Clone());
-        }
-
-        helper->saved_items_changed_callbacks_.Notify(helper->saved_items_);
-      },
-      this);
-  service_->GetAllPlaylistItems(std::move(on_get_all_items));
+  bool should_notify = false;
+  base::ranges::for_each(service_->GetAllPlaylistItems(),
```

Comment:
Just curious - any advantage to using `base::ranges::for_each` over `for (... : ...)`? Is it a style preference (as it lets us write code in a more functional style)?

Reply:
Yeah, for now and here both are equivalent, especially when we're iterating through all the given vector. for_each wins range-base for loop only when we want to specify start/end point that are not ::begin(), end().

But when C++20 is fully enabled in our source code, thing would be changed, with std::view::filter and ranges combination. 

This code could be (might not be correct, though)


```c++

service_->GetAllPlaylistItems() | 
  std::view::filter([](const auto& item) { return item->page_source == GetWebContents()->GetVisibleURL(); }) |
  std::view::for_each([](const auto& item){ saved_items_.push_back(item->Clone());
                           should_notify = true; });
```

More functional, exactly like you said.